packages/astro-angular/src/server.ts

```diff
@@ -1,35 +1,98 @@
 import 'zone.js/dist/zone.js';
-import { renderApplication } from '@angular/platform-server';
-import type { ɵComponentType } from '@angular/core';
+import type {
+  ComponentMirror,
+  Provider,
+  ɵComponentType as ComponentType,
+} from '@angular/core';
+import {
+  ApplicationRef,
+  InjectionToken,
+  reflectComponentType,
+} from '@angular/core';
+import {
+  BEFORE_APP_SERIALIZED,
+  renderApplication,
+} from '@angular/platform-server';
+
+const ANALOG_ASTRO_STATIC_PROPS = new InjectionToken<{
+  props: Record<string, unknown>;
+  mirror: ComponentMirror<unknown>;
+}>('Static props token w/ mirror', {
+  factory() {
+    return { props: {}, mirror: {} as ComponentMirror<unknown> };
+  },
+});
 
 function check(
-  Component: ɵComponentType<unknown>,
+  Component: ComponentType<unknown>,
   _props: Record<string, unknown>,
   _children: unknown
 ) {
-  return !!Component['ɵcmp'];
+  return !!reflectComponentType(Component);
 }
 
+// Run beforeAppInitialized hook to set Input on the ComponentRef
+// before the platform renders to string
+const STATIC_PROPS_HOOK_PROVIDER: Provider = {
```

Comment:
nit/question: Why HOOK_PROVIDER instead of just PROVIDER?

Reply:
It is mainly for the clear intent of the provider. 
- It is a `PROVIDER`
- It is to provide a `BEFORE_APP_INITIALIZED` hook -> `HOOK`
- The hook is to setup static props for the component -> `STATIC_PROPS`

Reply:
Got it. Thanks